### PR TITLE
Add purposes to preferences screen 

### PR DESCRIPTION
**Pull request by mklucsarits**
There are still several TODOs for the UI here.

### CODE REVIEW

File: src/components/Preferences.js [hunk 1]
```diff
@@ -1,12 +1,94 @@
...
+    };
+
+    componentWillMount() {
+        if (!this.state.purposes) {
+            fetchVendorList().then((json) => {
+                // Filter purposes to those used by Fandom
+                const purposes = json.purposes.filter(purpose => (this.props.allPurposes.indexOf(purpose.id) >= 0));
+                // Filter vendors to those used by Fandom
+                const vendors = json.vendors.filter(vendor => (this.props.allVendors.indexOf(vendor.id) >= 0));
+                this.setState({ purposes });
```
Comment (JoshuaRogan) [inline src/components/Preferences.js]: Should this be both purposes and vendors?

File: src/components/PreferencesSection.js [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
+    toggleIsExpanded() {
+        const { isExpanded } = this.state;
+        this.setState({ isExpanded: !isExpanded });
+        this.forceUpdate();
+    }
+
     render(props, state) {
 		const { heading, isEnabled } = props;
-		const { isExpanded } = state;
+        const { isExpanded } = this.state;
```
Comment (JoshuaRogan) [inline src/components/PreferencesSection.js]: If we are going to use `this.state` we should probably remove `state` from the method signature. Both work in preact but we should be consistent.

File: src/components/Preferences.js (hunk 1 above)
Comment (mklucsarits) [reply src/components/Preferences.js]: Yeah, I'm getting there. This file is still very WIP.

File: src/components/PreferencesSection.js (hunk 2 above)
Comment (mklucsarits) [reply src/components/PreferencesSection.js]: Good point

File: src/components/Preferences.js (hunk 1 above)
Comment (JoshuaRogan) [reply src/components/Preferences.js]: Sounds good. Overall, I'm on board with this approach.

File: src/components/ScreenOne.scss [hunk 3]
```diff
@@ -6,65 +6,82 @@
     }
 }
 
+.dialog {
+    background-color: white;
+    border-top-left-radius: 4px;
+    border-top-right-radius: 4px;
+    display: flex;
+    flex-direction: column;
+    font-family: Rubik, "Helvetica Neue", Helvetica, Arial, sans-serif !important;
```
Comment (vforge) [inline src/components/ScreenOne.scss]: There's a variable for it now

Comment (mklucsarits) [reply src/components/ScreenOne.scss]: It was actually duplicate code that I now removed.

**Pull request merged**